### PR TITLE
Follow selection, track positions

### PR DESCRIPTION
**Pull request by ManuelHentschel**
Includes the features mentioned in #76:
- Allow slices with multiple cursors. The flowr API already seems to support this, so it's just a matter of extracting the right info from `editor.selections`
- Auto-update the slice to follow the current selection. This can be toggled on/off.
- Allow the user to set "markers", which are used as slice criteria. The user can then edit the code to clean up dependencies and see live updates of the slice.
- Use `vscode.TextDocumentContentProvider` to output reconstructed slices. This way the contents of the slice can be updated continuously without opening new temporary documents (particularly useful for the previous two points)

Furthermore adds the slice display option `tokens`, which highlights the individual tokens from a slice. This felt like a nice feature to have when there is a lot going on in individual lines.

The following refactoring was done to make the changes above easier:
- (Remove activation events from `package.json`. The extension activates automatically when commands are called, and as far as I can see, these are the only ways to interact with the extension at the moment.)
- Move shared logic of `internal-session.ts` and `server-session.ts` into a separate file.
- Separate the slice retrieval and display functionality.
- Create separate decoration types for each selection/position slicer.
- Replace some exported, global objects by get functions. That way it's easier to initialize objects on demand and not worry about possibly undefined objects in other functions.

Some points to consider:
- New decos are not user-configurable yet. Since there are so many different ways to highlight things, I did not want to invest too much energy before discussing this. Also not really a breaking problem.
- The single-time slice operations are now part of the selection-slicer (updating it on demand, instead of on selection change)
- The decos of tracked positions are updated on config change. For selection slices this felt like disproportionate effort.
- I did not test with the server backend, haven't looked into that yet
- I tried to follow your coding conventions, but feel free to fix

Feel free to modify directly on this branch or let me know if you'd like me to change things.

### CODE REVIEW

File: src/slice.ts [hunk 1]
```diff
@@ -71,9 +98,34 @@ export async function displaySlice(editor: vscode.TextEditor, sliceElements: { i
...
+		}),
+		tokenSlice: vscode.window.createTextEditorDecorationType({
+			backgroundColor: 'green',
+		}),
+		slicedPos: vscode.window.createTextEditorDecorationType({
+			before: {
+				color:           'white',
+				contentText:     '->',
+				backgroundColor: 'green',
+				border:          '2px solid green',
```
Comment (Ellpeck) [inline src/slice.ts:120]: This color is quite a lot. Maybe we should make this customizable, and/or base it on the active theme if possible?

**Review by Ellpeck — CHANGES_REQUESTED** [review]
This looks really good! I requested some small changes - well, one change and one little nitpick, but I'm excited for this to be merged.

File: .vscode/settings.json [hunk 2]
```diff
@@ -11,5 +11,9 @@
     // run eslint when saving to format
     "eslint.format.enable": true,
     "editor.formatOnSave": true,
-    "eslint.enable": true
-}
+    "eslint.enable": true,
+    "[typescript]": {
```
[the remark this answers is not in the document]
Comment (ManuelHentschel) [reply .vscode/settings.json:15]: I had prettier installed and (by default) configured as default code formatter. As a consequence, these workspace settings caused everything to be formatted incorrectly (and highlighted as error) on every save. This was a leftover from me trying to fix the issue. I added an explicit indication of the eslint formatter to the config now, which I assume is the one you use.

File: src/slice.ts (hunk 1 above)
Comment (ManuelHentschel) [reply src/slice.ts]: The background color is now configurable and can be based on the current theme by using webview styling syntax.

**Review by Ellpeck — APPROVED** [review]

**Review by EagleoutIce — APPROVED** [review]
Amazing! Thank you very, very much!

**Pull request merged**